Jcszr6dt-6 Edycja danych z poziomu konsoli

raffalm opened the pull request.

raffalm committed:
Add JSON file and RideRepository

raffalm committed:
Ride - Edit name comleted

Add block //Name edit in EditRide method
modified:   PlanAndRide.BusinessLogic/RideRepository.cs

raffalm committed:
Ride-Edit date completed

modified:   PlanAndRide.BusinessLogic/Ride.cs
modified:   PlanAndRide.BusinessLogic/RideRepository.cs
modified:   PlanAndRide.GUI/data.json

raffalm committed:
Ride-Edit description completed

raffalm committed:
Ride-Edit ShareRide and IsPrivate completed

raffalm committed:
Ride - Route editing complete

raffalm committed:
Refactoring - extract methods

Edit Ride and Route data.
modified:   PlanAndRide/PlanAndRide.BusinessLogic/Ride.cs
modified:   PlanAndRide/PlanAndRide.BusinessLogic/RideRepository.cs

wojciechOlszewski1 commented on PlanAndRide/PlanAndRide.BusinessLogic/RideRepository.cs:
```diff
@@ -0,0 +1,317 @@
+
+        public static void EditRide(string rideName)
+        {
+            var myRide = rides.FirstOrDefault(r => r.Name == rideName);
```
Czy nazwa przejazdu jest unikatową wartością? jeżeli tak to lepiej użyć SingleOrDefault. FirstOrDefault szuka nam pierwszego wystąpienia i jeżeli będzie wiele wystąpień w tym przypadku wiele przejazdów o takiej samej nazwie to zwróci nam pierwszy jaki odnajdzie. W przypadku SingleOrDefault jeżeli będzie występować więcej niż jeden przejazd o danej nazwie wtedy rzuci błędem

wojciechOlszewski1 commented on PlanAndRide/PlanAndRide.BusinessLogic/RideRepository.cs:
```diff
@@ -0,0 +1,317 @@
+                return;
+            }
+
+            EditRideName(myRide);
```
logika edycji w repozytorium powinna być niezależna od UI jakie jest wykorzystywane do zbierania informacji co należy edytować. Teraz repozytorium jest bardzo powiązane z aplikacją konsolową. Metoda edycji w repozytorium powinna przyjmować tylko obiekt który chcemy edytować. Tzn w części UI czyli w Waszym projekcie konsolowym zbieramy wymagania i zmiany do istniejącego projektu. Taki nowy obiekt przesyłamy do repozytorium. W repozytorium robimy tylko update starych informacji.

raffalm replied:
Pomysł jest następujący:
- metody odpowiedzialne za wyszukanie i edycję obiektu przenosimy na stronę UI ( czyli EditRide(string rideName etc.) )
- po stronie UI, po wyszukaniu obiektu do edycji tworzymy klona
- użytkownik pracuje na nowym obiekcie (klonie) i edytuje interesujące go wartości
- po zakończeniu edycji "klona" przesyłamy do repozytorium i nadpisujemy "stary" obiekt

Co o tym sądzisz?

wojciechOlszewski1 commented on PlanAndRide/PlanAndRide.BusinessLogic/RideRepository.cs:
```diff
@@ -0,0 +1,317 @@
+        public static void EditRide(string rideName)
+        {
+            var myRide = rides.FirstOrDefault(r => r.Name == rideName);
+            if (myRide == null)
```
Zabezpieczenie przed nulem, fajnie że jest! :)

raffalm committed:
Merge branch 'main' into jcszr6dt-6-edit-data

raffalm committed:
Ride update logic moved to UI side

Refactor - change methods name
Changes to be committed:
	modified:   PlanAndRide/PlanAndRide.BusinessLogic/RideRepository.cs
	new file:   PlanAndRide/PlanAndRide.GUI/RideService.cs

raffalm committed:
Refactor - user language->PL, change logic YesNo settings

modified:   PlanAndRide/PlanAndRide.GUI/RideService.cs

raffalm commented:
Logika zbierania danych do edycji przeniesiona do UI :)
